### DISCUSSION TITLE
Can the browser of skywaking v9.0.0  record the client IP

### DISCUSSION THREAD

**Discussion by czxin788** [apache/skywalking#8925] [Q&A] [answered]
Can the browser of skywaking v9.0.0  record the client IP?
![image](https://user-images.githubusercontent.com/13861904/164579618-d7374c94-c2cf-4334-b04d-dd5befa211f2.png)

**Comment by wu-sheng** [comment, marked as answer]
This is a service, not an IP. Service means what would this app do. Collecting user information is sensitive, I am not sure. This is different from collecting server IP address from law perspective.

**Comment by czxin788** [comment]
I found pinpoint can collect client ip, I think it is usefull.
![image](https://user-images.githubusercontent.com/13861904/164584312-5d16ea66-aba6-4c16-b4b9-2f8610ed9991.png)

**Reply by wu-sheng** [reply]
Again and again,  what other projects or commercial products did is their choice. I have mentioned multiple times, that this could prove nothing.
Open source is for people working together, if you want something, take action, send a proposal, then a pull request.

**Comment by Superskyyy** [comment]
Why would you need a specific IP address anyway... also of course, likely violates user privacy, I think general geolocation and browser version info could be safer and more helpful in problem diagnosis.
